Controller/SecurityController.php

```diff
@@ -38,15 +48,20 @@ public function loginAction(Request $request)
         }
 
         // last username entered by the user
-        $lastUsername = (null === $session) ? '' : $session->get(SecurityContextInterface::LAST_USERNAME);
+        $lastUsername = (null === $session) ? '' : $session->get($lastUsernameKey);
 
-        $csrfToken = $this->has('form.csrf_provider')
-            ? $this->get('form.csrf_provider')->generateCsrfToken('authenticate')
-            : null;
+        if ($this->has('security.csrf.token_manager')) {
+            $csrfToken = $this->get('security.csrf.token_manager')->getToken('authenticate');
```

Comment:
should be ``->getToken('authenticate')->getValue()``. ``getToken`` returns a CsrfToken object, not a string